com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetectAppsToRenameStep.java

```diff
@@ -0,0 +1,130 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import com.sap.cloud.lm.sl.cf.core.model.BlueGreenApplicationNameSuffix;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.ImmutableCloudApplication;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import javax.inject.Named;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Named("detectAppsToRenameStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class DetectAppsToRenameStep extends SyncFlowableStep {
+
+    private static final String OLD_SUFFIX = BlueGreenApplicationNameSuffix.OLD.asSuffix();
+    private static final String NEW_SUFFIX = BlueGreenApplicationNameSuffix.NEW.asSuffix();
+
+    @Override
+    protected StepPhase executeStep(ExecutionWrapper execution) {
+        StepsUtil.setAppsToUndeploy(execution.getContext(), Collections.emptyList());
```

Comment:
Because the undeploy apps call activity follows this step and it needs this variable. It's cleaner to set it here, rather than in each of the execution flows (no deployed mta / no keepExistingAppNames flag / error)

Reply:
But it would be set anyway right???